MembraneRTC/src/main/java/org/membraneframework/rtc/media/SoundDetection.java

```diff
@@ -0,0 +1,116 @@
+package org.membraneframework.rtc.media;
+
+
+import android.media.AudioFormat;
+import android.media.AudioRecord;
+import android.media.MediaRecorder;
+
+import java.util.Timer;
+import java.util.TimerTask;
+
+import timber.log.Timber;
+
+public class SoundDetection {
+    private Timer timer;
+    private AudioRecord audioRecord;
+    private OnSoundDetectedListener onSoundDetectedListener;
+    private int bufferSize;
+    public boolean isRecording = false;
+    private final int volumeThreshold = -100;
+    private final int monitorInterval = 1;
+    private final int samplingRate = 22050;
+
+    public void start() throws SecurityException {
+        start(monitorInterval, samplingRate, volumeThreshold);
+    }
+
+    public void start(int monitorInterval, int samplingRate, int volumeThreshold) throws SecurityException {
```

Comment:
it's a public API, but it's not documented, you should document those class, functions and arguments